core/src/main/scala/org/apache/spark/deploy/SparkSubmit.scala

```diff
@@ -1072,14 +1072,29 @@ object SparkSubmit extends CommandLineUtils with Logging {
    * Return whether the given primary resource requires running python.
    */
   private[deploy] def isPython(res: String): Boolean = {
-    res != null && res.endsWith(".py") || res == PYSPARK_SHELL
+    if (res == null) {
+      return false
+    }
+    if (res == PYSPARK_SHELL) {
+      return true
+    }
+    val uri = new java.net.URI(res)
```

Comment:
I think the problem here is that it wouldn't work with local paths that has special characters disallowed in URI. See `PythonRunner` or `RRunner`.

Can we just do something like `Try(new java.net.URI(res).getPath).getOrElse(res).endsWith(".py")`? We will also have to have a test case.

Reply:
@HyukjinKwon Would this be a good place to add a test case? -> https://github.com/apache/spark/blob/master/core/src/test/scala/org/apache/spark/deploy/PythonRunnerSuite.scala

Good point about the local paths... I'll take care of that.

Reply:
Yes, I think you should add a test in places under `org.apache.spark.deploy` such as `SparkSubmitSuite` and `PythonRunnerSuite` as you pointed out.

Reply:
@HyukjinKwon I am handling the possible URI syntax exception now and also extracted the common code into a separate private function with a comment. Let me know whether that is more what you had in mind.

Gonna provide a test case later on.

This issue should be labelled SPARK_SUBMIT I think? Or can I add the label by changing the title? I noticed the square brackets allover the place.